internal/provider/kubernetes/controller.go

```diff
@@ -1249,6 +1275,22 @@ func (r *gatewayAPIReconciler) watchResources(ctx context.Context, mgr manager.M
 		return err
 	}
 
+	serviceImportFound := r.mcsServiceImportFound(mgr)
+	if !serviceImportFound {
+		r.log.Info("ServiceImport CRD not found, skipping ServiceImport watch")
+	}
+
+	// Watch ServiceImport CRUDs and process affected *Route objects.
+	if serviceImportFound {
+		if err := c.Watch(
+			source.Kind(mgr.GetCache(), &mcsapi.ServiceImport{}),
+			&handler.EnqueueRequestForObject{},
```

Comment:
lets use `handler.EnqueueRequestsFromMapFunc(r.enqueueClass)` here instead, similar to what is done for all other `Watch` calls, might require a `rebase`, since this change was done recently